src/jit/codegenarm64.cpp

```diff
@@ -6536,161 +6432,293 @@ CodeGen::genIntrinsic(GenTreePtr treeNode)
+                //             ldr     x2, [x0]
+                //             ldr     w3, [x0, #8]
+                //
+                // When the first instruction has a loReg that is the same register 
+                // as the source register: addrReg,  we set deferLoad to true and
+                // issue the intructions in the reverse order:
+                //             ldr     w3, [x2, #8]
+                //             ldr     x2, [x2]
+
+                bool      deferLoad     = false;
+                emitAttr  deferAttr     = EA_PTRSIZE;
+                int       deferOffset   = 0;
+                int       remainingSize = structSize;
+                unsigned  structOffset  = 0;
+                var_types nextType      = type0;
+
+                // Use the ldp instruction for a struct that is exactly 16-bytes in size
+                //             ldp     x2, x3, [x0]
+                //
+                if (remainingSize == 2*TARGET_POINTER_SIZE)
+                {
+                    if (hasGCpointers)
+                    {
+                        // We have GC pointers, so use two ldr instructions
+                        //
+                        // We do it this  way because we can't currently pass or track 
+                        // two different emitAttr values for a ldp instruction.
+
+                        // Make sure that the first load instruction does not overwrite the addrReg.
+                        //
+                        if (loReg != addrReg)
+                        {
+                            emit->emitIns_R_R_I(INS_ldr, emitTypeSize(type0), loReg, addrReg, structOffset);
+                            emit->emitIns_R_R_I(INS_ldr, emitTypeSize(type1), hiReg, addrReg, structOffset + TARGET_POINTER_SIZE);
+                        }
+                        else 
+                        {
+                            assert(hiReg != addrReg);
+                            emit->emitIns_R_R_I(INS_ldr, emitTypeSize(type1), hiReg, addrReg, structOffset + TARGET_POINTER_SIZE);
+                            emit->emitIns_R_R_I(INS_ldr, emitTypeSize(type0), loReg, addrReg, structOffset);
+                        }
+                    }
+                    else
+                    {
+                        // Use a ldp instruction 
+
+                        emit->emitIns_R_R_R_I(INS_ldp, EA_PTRSIZE, loReg, hiReg, addrReg, structOffset);
+                    }
+                    remainingSize = 0;     // We completely wrote the 16-byte struct
+                }
+
+                regNumber curReg = loReg;
+                while (remainingSize > 0)
+                {
+                    if (remainingSize >= TARGET_POINTER_SIZE)
+                    {
+                        remainingSize -= TARGET_POINTER_SIZE;
+
+                        if ((curReg != addrReg) || (remainingSize == 0))
+                        {
```

Comment:
This is a minor suggestion - this is just confusing enough to warrant an additional comment. Also, you might consider reversing the condition, because if ((curReg == addrReg) && (remainingSize != 0)) seems (to me) to more clearly explain that we need to defer this load, because we can't overwrite addrReg if there's more to write.